src/Simulation/TargetDefinitions/Intrinsic/Checks.qs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Intrinsic {
+    open Microsoft.Quantum.Diagnostics;
+
+    /// # Summary
+    /// Checks that all qubits operated on are unique.
+    ///
+    /// # Description
+    /// Will fail if any qubits are repeated or null.
+    ///
+    /// # Input
+    /// ## qubits
+    /// The array of qubits to verify for uniqueness. In the controlled variant
+    /// the full list of qubits among targets and controls are verified to be unique.
+    @EnableTestingViaName("Test.TargetDefinitions.CheckQubitUniqueness")
+    internal operation CheckQubitUniqueness (qubits : Qubit[]) : Unit is Adj + Ctl{
```

Comment:
How come this is an operation? I believe it should always return the same result given the same input, right?

Reply:
Yeah, this can definitely be a function, or better yet, removed (fingers crossed).